src/stan/io/deserializer.hpp

```diff
@@ -0,0 +1,1978 @@
+   *   for a correlation
+   */
+  template <typename Ret, typename... Sizes,
+            require_not_matrix_t<Ret>* = nullptr,
+            require_not_std_vector_t<Ret>* = nullptr>
+  auto read_corr(Sizes... sizes) {
+    auto ret = read<Ret>(sizes...);
+    using stan::math::check_bounded;
+    using stan::math::value_of;
+    check_bounded<decltype(value_of(ret)), double, double>(
+        "deserializer", "Correlation value", value_of(ret), -1, 1);
+    return ret;
+  }
+
+  /**
+   * Specialization of `read_corr` for `std::vector` returns
+   * @tparam Ret The type to return.
+   * @tparam Sizes A parameter pack of integral types.
+   * @param sizes Pack of integrals to use to construct the return's type.
+   * @return Next correlation value.
+   * @throw std::runtime_error if the value is not valid
+   *   for a correlation
+   */
+  template <typename Ret, typename... Sizes,
+            require_std_vector_t<Ret>* = nullptr>
+  auto read_corr(size_t vecsize, Sizes... sizes) {
+    std::decay_t<Ret> ret;
+    ret.reserve(vecsize);
+    for (size_t i = 0; i < vecsize; ++i) {
+      ret.emplace_back(this->read_corr<value_type_t<Ret>>(sizes...));
+    }
+    return ret;
+  }
+
+  /**
+   * Specialization of `read_corr` for matrix returns
+   * @tparam Ret The type to return.
+   * @param k the rows and column size of the return matrix.
+   * @throw std::runtime_error if the value is not valid
+   *   for a correlation
+   */
+  template <typename Ret, require_matrix_t<Ret>* = nullptr>
+  auto read_corr(size_t k) {
+    auto ret = read<Ret>(k, k);
+    using stan::math::check_corr_matrix;
+    using stan::math::value_of;
+    check_corr_matrix("stan::math::corr_matrix", "Constrained matrix",
+                      value_of(ret));
+    return ret;
+  }
+
+  /**
+   * Return the next object transformed to be a (partial)
+   * correlation between -1 and 1, incrementing the specified
+   * reference with the log of the absolute Jacobian determinant.
+   *
+   * <p>See <code>stan::math::corr_constrain(T,T&)</code>.
+   *
+   * @tparam Ret The type to return.
+   * @tparam Jacobian Whether to increment the log of the absolute Jacobian
+   * determinant of the transform.
+   * @tparam LP Type of log probability.
+   * @tparam Sizes A parameter pack of integral types.
+   * @param sizes Pack of integrals to use to construct the return's type.
+   * @param lp The reference to the variable holding the log
+   * probability to increment.
+   * @return The next scalar transformed to a correlation.
+   */
+  template <typename Ret, bool Jacobian, typename LP, typename... Sizes,
+            require_not_std_vector_t<Ret>* = nullptr,
+            require_not_matrix_t<Ret>* = nullptr>
+  auto read_corr(LP& lp, Sizes... sizes) {
+    using stan::math::corr_constrain;
+    if (Jacobian) {
+      return corr_constrain(this->read<Ret>(sizes...), lp);
+    } else {
+      return corr_constrain(this->read<Ret>(sizes...));
+    }
+  }
```

Comment:
Similarly this can be removed if there is no corr constraint for vectors in the language (and I don't think there is)

Reply:
Is there a corr scalar type even?

Reply:
Also this one is for just scalars since the requires on it stop return types of std vectors or matrics/vectors

Reply:
Yeah we can definitely clean this one up too. There is no scalar corr. I imagine maybe this was once used for partial correlations or something (constrained [-1, 1]) but upper/lower handle that now.